lib/block/header.go

```diff
@@ -20,13 +20,13 @@ type Header struct {
 	// TODO smart contract fields
 }
 
-func NewBlockHeader(round round.Round, txRoot string) *Header {
+func NewBlockHeader(basis voting.Basis, txRoot string) *Header {
 	return &Header{
-		PrevBlockHash:    round.BlockHash,
+		PrevBlockHash:    basis.BlockHash,
```

Comment:
Isn't that super inconsistent though ? `basis.BlockHash` => previous block hash, but `basis.Height` => current block height.

Reply:
@Geod24 Yes right. I think it seems to be confusing. But it does not related this change.
We can handle it after this pr :)

Reply:
That's right